lib/filewatch/helper.rb

```diff
@@ -0,0 +1,73 @@
+# code downloaded from Ruby on Rails 4.2.1
+# https://raw.githubusercontent.com/rails/rails/v4.2.1/activesupport/lib/active_support/core_ext/file/atomic.rb
+require 'fileutils'
+
+class File
+  # Write to a file atomically. Useful for situations where you don't
+  # want other processes or threads to see half-written files.
+  #
+  #   File.atomic_write('important.file') do |file|
+  #     file.write('hello')
+  #   end
+  #
+  # If your temp directory is not on the same filesystem as the file you're
+  # trying to write, you can provide a different temporary directory.
+  #
+  #   File.atomic_write('/data/something.important', '/data/tmp') do |file|
+  #     file.write('hello')
+  #   end
+  def self.atomic_write(file_name, temp_dir = Dir.tmpdir)
+    require 'tempfile' unless defined?(Tempfile)
+    require 'fileutils' unless defined?(FileUtils)
+
+    temp_file = Tempfile.new(basename(file_name), temp_dir)
+    temp_file.binmode
+    return_val = yield temp_file
+    temp_file.close
+
+    if File.exist?(file_name)
+      # Get original file permissions
+      old_stat = stat(file_name)
+    else
+      # If not possible, probe which are the default permissions in the
+      # destination directory.
+      old_stat = probe_stat_in(dirname(file_name))
+    end
+
+    # Overwrite original file with temp file
+    FileUtils.mv(temp_file.path, file_name)
```

Comment:
If file_name and temp_dir are on different file systems, then this move will be a `copy` instead of a `rename`. A copy is not atomic.

Recommend writing to a file near the `file_name` - something like `file_name + ".new"` which you can write, sync, then rename (rename will be atomic, iirc)